metrics/scaling/k8s_scale.sh

```diff
@@ -104,12 +104,17 @@ EOF
 		cpu_idle=${cpu_idle:-0}
 		mem_free=${mem_free:-0}
 
+		# bash arrays don't handle hyphens in the index as it treats
+		# a name like "azure-east-1" as 0-0-1, so remove any hyphens in the
+		# name, will work as long as hyphens aren't the only difference
+		node_index=$(tr -d [=-=] <<< $node)
+
 		# If this is the 0 node instance, store away the base memory value
 		if [ $n_pods -eq 0 ]; then
-			node_basemem[$node]=$mem_free
+			node_basemem[$node_index]=$mem_free
```

Comment:
Hmm, I wonder if there is a deeper root cause or an alternative method here... I had a play locally (as bash associative arrays are not exactly my strong point ;-) )... thoughts:
- initial thought was `can we fix this by "'ting the $node`?
- but then I had a play, and found that we declare `node_basemen` with `declare -a`, whereas we are using it as an associative array I think, so should we be using `declare -A` instead? Maybe that will just make the string indexing work?, or maybe in combination with quoting the `$node` expansion, will make it work?